### ISSUE TITLE
Exception when returning from parent page

### ISSUE THREAD

**Issue by be-swarm** [afriscic/BarcodeScanning.Native.Maui#4]
If the control is placed in a secondary page, when you return to the main page an exception is thrown.

Step to reproduce:
Create a page with your control.
register it in di (optional)
 builder.Services.AddSingleton<YourPage>();

From main page:
await Shell.Current.GoToAsync($"{nameof(YourPage}");

From your page
await Shell.Current.GoToAsync($"..");

Exception:
System.ObjectDisposedException: 'Cannot access a disposed object.
Object name: 'AndroidX.Camera.View.PreviewView'.'

**Comment by afriscic** [comment]
Try to add it in DI as transient and/or override OnDisappearing and set CameraEnabled to false to close the camre before you move out of the page.

**Comment by tolgraven** [comment]
This should be fixed by the PR my colleague got merged (thanks for the quick turnaround @afriscic), so try the latest nuget @be-swarm

**Comment by geertgeerits** [comment]
Same problem here.
From main page to secondary page: await Navigation.PushAsync(new PageScanNT());
Using the back button of the page to return, than I get also that error: Cannot access a disposed object.
I did set CameraEnabled to false in the OnDisappering event but still that error.
BarcodeScanning.Native.Maui version 1.0.2
Visual Studio 2022 version 17.8.1

**Comment by afriscic** [comment]
OK, I have finally managed to consistently replicate this error from your scenario and found the culprit.
Should be fixed in 1.0.3 and I'll mark this issue as resolved.

**Comment by Ghostbird** [comment]
I'm having this issue in 1.3.1:

```stacktrace
Java.Interop.JniPeerMembers.AssertSelf (/home/user/workspace/app/.meteor/sources/xamarin/java.interop/bec0326a925ff888d96cb433f5049000a6d32a33/src/Java.Interop/Java.Interop/JniPeerMembers.cs:153)
Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeVirtualObjectMethod (/home/user/workspace/app/.meteor/sources/xamarin/java.interop/bec0326a925ff888d96cb433f5049000a6d32a33/src/Java.Interop/Java.Interop/JniPeerMembers.JniInstanceMethods_Invoke.cs:858)
AndroidX.Camera.View.CameraController.EnableTorch (AndroidX.Camera.View.CameraController.cs:1654)
BarcodeScanning.CameraViewHandler.Stop (:-1)
Java.Lang.Thread.RunnableImplementor.Run (/home/user/workspace/app/.meteor/sources/xamarin/xamarin-android/f68622cf6b97fa23cc3d3105a52ef5b2e31d52a1/src/Mono.Android/Java.Lang/Thread.cs:36)
Java.Lang.IRunnableInvoker.n_Run (/home/user/workspace/app/.meteor/sources/xamarin/xamarin-android/f68622cf6b97fa23cc3d3105a52ef5b2e31d52a1/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Java.Lang.IRunnable.cs:84)
Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V (/home/user/workspace/app/.meteor/sources/xamarin/xamarin-android/f68622cf6b97fa23cc3d3105a52ef5b2e31d52a1/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:26)
```
